### PR TITLE
Flow type fixes

### PR DESCRIPTION
**Pull request by Ashoat**
Explanations inline

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -140,7 +140,7 @@ export type UserCredentials = {|
 export function getInternetCredentials(
   server: string,
   options?: Options
-): Promise<UserCredentials> {
+): Promise<false | UserCredentials> {
```
Comment (Ashoat) [inline index.js]: [Can resolve as `false`](https://github.com/oblador/react-native-keychain/blob/master/RNKeychainManager/RNKeychainManager.m#L427)

File: index.js [hunk 2]
```diff
@@ -235,7 +235,7 @@ export function resetGenericPassword(
  * @return {Promise} Resolves to `{ server, username, password }` if approved and
  * `false` if denied and throws an error if not supported on platform or there's no shared credentials
  */
-export function requestSharedWebCredentials(): Promise<SharedWebCredentials> {
+export function requestSharedWebCredentials(): Promise<false | SharedWebCredentials> {
```
Comment (Ashoat) [inline index.js]: [Can resolve as `false`](https://github.com/oblador/react-native-keychain/blob/master/RNKeychainManager/RNKeychainManager.m#L476)

File: index.js [hunk 3]
```diff
@@ -256,7 +256,7 @@ export function requestSharedWebCredentials(): Promise<SharedWebCredentials> {
 export function setSharedWebCredentials(
   server: string,
   username: string,
-  password: string
+  password: ?string
```
Comment (Ashoat) [inline index.js]: A `NULL` `password` passed to [`SecAddSharedWebCredential`](https://github.com/oblador/react-native-keychain/blob/master/RNKeychainManager/RNKeychainManager.m#L486) is used to [remove a shared password](https://developer.apple.com/documentation/security/1617986-secaddsharedwebcredential?language=objc)

**Comment by vonovak** [comment]
thanks!

**Pull request merged**